dali/operators/image/remap/remap.cuh

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_IMAGE_REMAP_REMAP_CUH_
+#define DALI_OPERATORS_IMAGE_REMAP_REMAP_CUH_
+
+#include "dali/core/cuda_stream_pool.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/pipeline/operator/common.h"
+#include "include/dali/core/static_switch.h"
+#include "dali/kernels/dynamic_scratchpad.h"
+
+namespace dali {
+namespace remap {
+namespace detail {
+
+template<typename T>
+std::enable_if_t<std::is_floating_point<T>::value>
+__global__
+shift_pixel_origin_per_batch(T **data, const size_t *sample_sizes, size_t n_samples,
+                             T shift_value) {
+  for (int sample_id = blockIdx.y;
+       sample_id < n_samples;
+       sample_id += gridDim.y) {
+    for (int component_id = blockIdx.x * blockDim.x + threadIdx.x;
+         component_id < sample_sizes[sample_id];
+         component_id += blockDim.x * gridDim.x) {
+      data[sample_id][component_id] += shift_value;
+    }
+  }
+}
+
+
+template<typename T>
+void
+invoke_kernel_per_batch(T **data_buffers, const size_t *sample_sizes, int n_samples, T shift_value,
+                        cudaStream_t stream) {
+  static constexpr int kBlockSizeX = 256;
+  static constexpr float kOneOverBlockSize = 1.f / static_cast<float>(kBlockSizeX);
+  dim3 block_size(kBlockSizeX);
+  dim3 grid_size(32, 32);
+  shift_pixel_origin_per_batch<<<grid_size, block_size, 0, stream>>>
+          (data_buffers, sample_sizes, n_samples, shift_value);
+}
+
+
+template<typename StorageBackend, typename T, int ndims>
+T **GetGpuAccessibleTensors(TensorListView<StorageBackend, T, ndims> tlv,
+                            dali::kernels::DynamicScratchpad &ds, cudaStream_t stream) {
+  using DataType = T *;
+  auto data_size = tlv.num_samples() * sizeof(DataType);
+  DataType *ret = ds.template AllocatePinned<DataType>(data_size);
+  cudaMemcpyAsync(ret, tlv.data.data(), data_size, cudaMemcpyDefault, stream);
+  return ret;
+}
+
+
+size_t *GetGpuAccessibleSampleSizes(size_t *sample_sizes, size_t n_samples,
+                                    dali::kernels::DynamicScratchpad &ds, cudaStream_t stream) {
+  auto data_size = n_samples * sizeof(size_t);
+  auto ret = ds.template AllocatePinned<size_t>(data_size);
+  CUDA_CALL(cudaMemcpyAsync(ret, sample_sizes, data_size, cudaMemcpyDefault, stream));
+  return ret;
+}
```

Comment:
I guess you could just use `scratchpad.ToContiguousGPU`.

Reply:
TBH, I'm not fond of using functions that aren't self-explanatory and there's no documentation for them. In this case I'd rather use 3 LOCs instead of 1 undocumented ;)